teradata/datadog_checks/teradata/check.py

```diff
@@ -0,0 +1,176 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import json
+import time
+from contextlib import closing, contextmanager
+
+try:
+    import teradatasql
+
+    TERADATASQL_IMPORT_ERROR = None
+except ImportError as e:
+    teradatasql = None
+    TERADATASQL_IMPORT_ERROR = e
+
+
+from datadog_checks.base import AgentCheck, is_affirmative
+from datadog_checks.base.constants import ServiceCheck
+from datadog_checks.base.utils.db import QueryManager
+
+from .config_models import ConfigMixin
+from .queries import COLLECT_RES_USAGE, DEFAULT_QUERIES
+
+SERVICE_CHECK_CONNECT = 'can_connect'
+SERVICE_CHECK_QUERY = 'can_query'
+
+
+class TeradataCheck(AgentCheck, ConfigMixin):
+    __NAMESPACE__ = 'teradata'
+
+    def __init__(self, name, init_config, instances):
+        super(TeradataCheck, self).__init__(name, init_config, instances)
+
+        self._connect_params = None
+        self._connection = None
+
+        self._tags = [
+            'teradata_server:{}'.format(self.instance.get('server')),
+            'teradata_port:{}'.format(self.instance.get('port', 1025)),
+        ]
+        self._tags.extend(self.instance.get('tags', []))
+
+        manager_queries = []
+        if is_affirmative(self.instance.get('collect_res_usage', False)):
+            manager_queries.extend(COLLECT_RES_USAGE)
+        else:
+            manager_queries.extend(DEFAULT_QUERIES)
```

Comment:
let's copy the default and append based on options

Reply:
Currently you are collecting either RES_USAGE or DEFAULT_QUERIES.

Reply:
good catch! i've fixed that now